packages/editor-ui/src/utils/nodeTypesUtils.ts

```diff
@@ -305,3 +313,247 @@ export const hasOnlyListMode = (parameter: INodeProperties): boolean => {
+	// Main authentication field has to be required
+	const isFieldRequired = mainAuthFiled ? isNodeParameterRequired(nodeType, mainAuthFiled) : false;
+	return mainAuthFiled && isFieldRequired ? mainAuthFiled : null;
+};
+
+// A field is considered main auth filed if:
+// 1. It is a credential dependency
+// 2. If all of it's possible values are used in credential's display options
+const findAlternativeAuthField = (
+	nodeType: INodeTypeDescription,
+	fields: INodeProperties[],
+): INodeProperties | null => {
+	const dependentAuthFieldValues: { [fieldName: string]: string[] } = {};
+	nodeType.credentials?.forEach((cred) => {
+		if (cred.displayOptions && cred.displayOptions.show) {
+			for (const fieldName in cred.displayOptions.show) {
+				dependentAuthFieldValues[fieldName] = (dependentAuthFieldValues[fieldName] || []).concat(
+					(cred.displayOptions.show[fieldName] || []).map((val) => (val ? val.toString() : '')),
+				);
+			}
+		}
+	});
+	const alternativeAuthField = fields.find((field) => {
+		let required = true;
+		field.options?.forEach((option) => {
+			if (!dependentAuthFieldValues[field.name].includes(option.value)) {
+				required = false;
+			}
+		});
+		return required;
+	});
+	return alternativeAuthField || null;
+};
+
+// Gets all authentication types that a given node type supports
+export const getNodeAuthOptions = (
+	nodeType: INodeTypeDescription | null,
+): NodeAuthenticationOption[] => {
+	if (!nodeType) {
+		return [];
+	}
+	let options: NodeAuthenticationOption[] = [];
+	const authProp = getMainAuthField(nodeType);
+	// Some nodes have multiple auth fields with same name but different display options so need
+	// take them all into account
+	const authProps = getNodeAuthFields(nodeType).filter((prop) => prop.name === authProp?.name);
+
+	authProps.forEach((field) => {
+		if (field.options) {
+			options = options.concat(
+				field.options.map((option) => ({
+					name: option.name,
+					value: option.value,
+					// Also add in the display options so we can hide/show the option if necessary
+					displayOptions: field.displayOptions,
+				})) || [],
+			);
+		}
+	});
+	return options;
+};
+
+export const getAllNodeCredentialForAuthType = (
+	nodeType: INodeTypeDescription | null,
+	authType: string,
+): INodeCredentialDescription[] => {
+	if (nodeType) {
+		return (
+			nodeType.credentials?.filter(
+				(cred) => cred.displayOptions?.show && authType in (cred.displayOptions.show || {}),
+			) || []
+		);
+	}
+	return [];
+};
+
+export const getNodeCredentialForSelectedAuthType = (
+	nodeType: INodeTypeDescription,
+	authType: string,
+): INodeCredentialDescription | null => {
+	const authField = getMainAuthField(nodeType);
+	const authFieldName = authField ? authField.name : '';
+	return (
+		nodeType.credentials?.find(
+			(cred) =>
+				cred.displayOptions?.show && cred.displayOptions.show[authFieldName]?.includes(authType),
+		) || null
+	);
+};
```

Comment:
Also, can we return early if `authField.name` is `undefined`?
```suggestion
export const getNodeCredentialForSelectedAuthType = (
	nodeType: INodeTypeDescription,
	authType: string,
) => {
	const authField = getMainAuthField(nodeType);
	const authFieldName = authField ? authField.name : '';
	return (
		nodeType.credentials?.find((cred) =>
			cred.displayOptions?.show?.[authFieldName]?.includes(authType),
		) || null
	);
};
```